src/domain/hooks/helpers/event-cache.helper.ts

```diff
@@ -0,0 +1,553 @@
+    ];
+  }
+
+  private onTransactionEventMessageConfirmation(
+    event: Extract<Event, { type: TransactionEventType.MESSAGE_CONFIRMATION }>,
+  ): Array<Promise<void>> {
+    // A new message confirmation affects:
+    // - the message itself
+    // - the messages associated to the Safe
+    return [
+      this.messagesRepository.clearMessagesByHash({
+        chainId: event.chainId,
+        messageHash: event.messageHash,
+      }),
+      this.messagesRepository.clearMessagesBySafe({
+        chainId: event.chainId,
+        safeAddress: event.address,
+      }),
+    ];
+  }
+
+  // Config Service events
+
+  private onConfigEventChainUpdate(
+    event: Extract<Event, { type: ConfigEventType.CHAIN_UPDATE }>,
+  ): Array<Promise<void>> {
+    // As the chains have been updated, we need to clear the memoized function cache.
+    if (this.isSupportedChainMemo.cache.clear) {
+      this.isSupportedChainMemo.cache.clear();
+    }
+    return [
+      this.chainsRepository.clearChain(event.chainId).then(() => {
+        // RPC may have changed
+        this.blockchainRepository.clearApi(event.chainId);
+        // Testnet status may have changed
+        this.stakingRepository.clearApi(event.chainId);
+        // Transaction Service may have changed
+        this.transactionsRepository.clearApi(event.chainId);
+        this.balancesRepository.clearApi(event.chainId);
+      }),
+    ];
+  }
+
+  private onConfigEventSafeAppsUpdate(
+    event: Extract<Event, { type: ConfigEventType.SAFE_APPS_UPDATE }>,
+  ): Array<Promise<void>> {
+    return [this.safeAppsRepository.clearSafeApps(event.chainId)];
+  }
+
+  private onTransactionEventSafeCreated(
+    event: Extract<Event, { type: TransactionEventType.SAFE_CREATED }>,
+  ): Array<Promise<void>> {
+    return [this.safeRepository.clearIsSafe(event)];
+  }
+
+  private _logSafeTxEvent(
+    event: Event & { address: string; safeTxHash: string },
+  ): void {
+    this.loggingService.info({
+      type: EventCacheHelper.HOOK_TYPE,
+      eventType: event.type,
+      address: event.address,
+      chainId: event.chainId,
+      safeTxHash: event.safeTxHash,
+    });
+  }
+
+  private _logTxEvent(
+    event: Event & { address: string; txHash: string },
+  ): void {
+    this.loggingService.info({
+      type: EventCacheHelper.HOOK_TYPE,
+      eventType: event.type,
+      address: event.address,
+      chainId: event.chainId,
+      txHash: event.txHash,
+    });
+  }
+
+  private _logMessageEvent(
+    event: Event & { address: string; messageHash: string },
+  ): void {
+    this.loggingService.info({
+      type: EventCacheHelper.HOOK_TYPE,
+      eventType: event.type,
+      address: event.address,
+      chainId: event.chainId,
+      messageHash: event.messageHash,
+    });
+  }
+
+  private _logEvent(event: Event): void {
+    this.loggingService.info({
+      type: EventCacheHelper.HOOK_TYPE,
+      eventType: event.type,
+      chainId: event.chainId,
+    });
+  }
+}
+
+@Module({
```

Comment:
What do you think about having a separate file called event-cache.module for this module? As it makes the module class more accessible and improves the readability of the code

Reply:
There's a pattern of including the `Module` in less "significant" [helpers](https://github.com/safe-global/safe-client-gateway/blob/refactor-hooks-repository/src/domain/swaps/contracts/decoders/gp-v2-decoder.helper.ts#L783), [repositories](https://github.com/safe-global/safe-client-gateway/blob/refactor-hooks-repository/src/routes/transactions/helpers/swap-order.helper.ts#L159), etc. However, I created a separate module in 9fb9f4ff1ae9ed9cd25837cc165356946b72e5b1 as I agree (`event-cache.helper.module.ts`).

We should reconsider other areas where we follow the aforementioned pattern as well.